src/plugins/console/public/lib/es/es.ts

```diff
@@ -58,6 +65,11 @@ export function send(
         jqXHR.responseText =
           "\n\nFailed to connect to Console's backend.\nPlease check the Kibana server is up and running";
       }
+      if (jqXHR.status === 400) {
+        jqXHR.responseText =
+          '400 - Bad request.\nPlease check the query format, spell errors, etc.';
+      }
+
```

Comment:
Ideally, we would return the error from the Elasticsearch response rather than a generic error. 

I tried to set `responseType: 'arraybuffer'` for all requests and parse the JSON response when `jqXHR.getResponseHeader('content-type') === 'application/json'`, but jQuery throws an error as it is appears to try to parse the response as JSON. 🤷 

https://github.com/elastic/kibana/pull/127867 is proposing replacing jQuery with Fetch for 8.3. We could consider waiting until that PR is merged as that might give us better support for binary responses.

Reply:
@nickpeihl Ok, good point. Let's wait for the jQuery replacement task first, it will also cause a conflict of changes in files.

Reply:
@nickpeihl Done. No additional checks for response errors are required, the message from the server is displayed by default:
<img width="1191" alt="Screenshot 2022-04-07 at 19 25 11" src="https://user-images.githubusercontent.com/3940730/162251543-a878090d-dd9c-4a9d-baaa-2de9350fd63c.png">